x-pack/plugins/security_solution/server/lib/sourcerer/routes/helpers.ts

```diff
@@ -14,8 +14,14 @@ export const findExistingIndices = async (
   Promise.all(
     indices
       .map(async (index) => {
+        const splitIndex = index.split('');
+        const isLeadingDash = splitIndex.shift() === '-';
+        let indexToQuery = index;
+        if (isLeadingDash) {
+          indexToQuery = splitIndex.join('');
+        }
         const searchResponse = await esClient.fieldCaps({
-          index,
+          index: indexToQuery,
```

Comment:
why are we even looping here? The fieldCaps api can take an array of indices https://github.com/elastic/elasticsearch-js/blob/main/src/api/types.ts#L341

Reply:
that's the type from https://github.com/elastic/elasticsearch-js/blob/main/src/api/api/field_caps.ts#L43

Reply:
https://github.com/elastic/elasticsearch-js/blob/main/src/api/types.ts#L2003

Reply:
If we do just one at a time, we can check the response length to know if there was a match. If we do more than one, we need to loop over each and do a string match on the response. however, the string match is complicated because wildcard. we tried it but could not get it to work.

Reply:
Can probably do it with some regex stuff but ya that makes sense, eventually 1 or none (I don't think the beat field descriptions are worth the overhead here really cc @XavierM )